src/spatialdata/_core/data_extent.py

```diff
@@ -362,6 +330,7 @@ def _compute_extent_in_coordinate_system(
     df = pd.DataFrame(corners.data, columns=corners.axis.data.tolist())
     points = PointsModel.parse(df, coordinates={k: k for k in axes})
     transformed_corners = transform(points, transformation).compute()
-    min_coordinates = transformed_corners.min().to_numpy()
-    max_coordinates = transformed_corners.max().to_numpy()
+    # Make sure min and max values are in the same order as axes
+    min_coordinates = transformed_corners.min()[list(axes)].to_numpy()
```

Comment:
This should fix the issue https://github.com/scverse/spatialdata/issues/335#issue-1842914360